### PR TITLE
monitoring: Add generic config generator using jinja templates

### PR DESCRIPTION
**Pull request by anmolbabu**
Add generic config generator using jinja templates

Signed-off-by: anmolbabu anmolbudugutta@gmail.com

### CODE REVIEW

**Comment by anmolbabu** [comment]
@nthomas-redhat @shtripat can you please review this patch. It adds a generic configuration generation technique to generate collectd configuration files using jinja templates. The templates will be maintained by sds-bridges(sds-specific) and node-agent(physical resources configuration)

**Comment by anmolbabu** [comment]
@brainfunked @r0h4n Please review

**Comment by anmolbabu** [comment]
TODO: Add UT's

File: bridge_common/monitoring/jinja2conf.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        self.template_path = templatePath
+        self.config_path = configPath
+        self.data = data
+
+
+    def generateConfigFile(self):
+        j2_env = Environment(loader=FileSystemLoader(os.path.dirname(self.template_path)))
+        template = j2_env.get_template(os.path.basename(self.template_path))
+        conf_str = template.render(self.data)
+        with open(self.config_path, 'w') as f:
```
Comment (nthomas-redhat) [inline bridge_common/monitoring/jinja2conf.py]: Handle exceptions all scenarios - open, environment, get_template etc

**Pull request closed without merging**